Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -414,7 +414,7 @@ inline HRESULT PopEndDraw() {
     template <typename Lambda> // Lambda takes the form HRESULT (*)(CGContextRef, ID2D1DeviceContext*)
     HRESULT Draw(_CGCoordinateMode coordinateMode, CGAffineTransform* additionalTransform, Lambda&& drawLambda);
 
-    HRESULT DrawGeometry(_CGCoordinateMode coordinateMode, ID2D1Geometry* pGeometry, CGPathDrawingMode drawMode);
+    HRESULT DrawGeometry(_CGCoordinateMode coordinateMode, ID2D1Geometry* pGeometry, CGPathDrawingMode drawMode, bool isPathGeometry);
```

Comment:
It's still required since a rectangle geometry is created without a particular fill mode. There's no way to set a particular default for simply an ID2D1RectangleGeometry (Similarly for others). An equivalent comparison would be creating a path, streaming a rectangle geometry into that path, and setting the fill mode on the geometrysink. I think the groups work better for this though.

As far as I know, you can't just assign a fill mode to a rectangle geometry or any normal ID2D1Geometry for that matter and have to do it through a group.